<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>677</width>
    <height>610</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="mainLayout">
    <item>
     <widget class="QWidget" name="leftContainerWidget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>NTrode Selection</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="1" column="1">
           <widget class="QPushButton" name="trainLFPStatisticsButton">
            <property name="text">
             <string>Train LFP Statistics</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QTableWidget" name="tableWidget">
            <property name="rowCount">
             <number>1</number>
            </property>
            <property name="columnCount">
             <number>3</number>
            </property>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <row/>
            <column>
             <property name="text">
              <string>NTrode Id</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Mean</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Std Dev</string>
             </property>
            </column>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="freezeSelectionButton">
            <property name="text">
             <string>Freeze Selection</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="trainingDurationLabel">
            <property name="text">
             <string>Training Duration (s)</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="trainingDurationSpinBox"/>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="trainingProgressLabel">
            <property name="text">
             <string>Training Time Remaining</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QProgressBar" name="trainingProgressBar"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Raspberry Pi Network Endpoint</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <property name="verticalSpacing">
           <number>-1</number>
          </property>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="stimDurationSpinBox"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Stim Duration (Âµs)</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QLineEdit" name="raspberryPiLineEdit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>udp://10.38.0.1:10000</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="testStimButton">
            <property name="text">
             <string>Test Stim</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="raspberryPiStatusLabel">
            <property name="text">
             <string>Status</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="rightContainerWidget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="rippleParamGroupBox">
         <property name="title">
          <string>Ripple Detection Parameters</string>
         </property>

         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="toolTip">
             <string>Ripple detection threshold in standard deviations
                                          above the
                                          mean.</string>
            </property>
            <property name="text">
             <string>Detection Threshold</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="rippleThreshold">
            <property name="maximum">
             <double>100000.0</double>
            </property>

           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Number of Channels</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="numActiveChannels">
            <property name="minimum">
             <number>1</number>
            </property>
           </widget>
          </item>

          <item row="2" column="0">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>Min Inter-Stim (ms)</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="minInterStim">
            <property name="minimum">
             <double>0.0</double>
            </property>
            <property name="maximum">
             <double>100000.0</double>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Max Stim Rate (Hz)</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QDoubleSpinBox" name="maxStimRate">
            <property name="minimum">
             <double>0</double>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Instantaneous Ripple Rate</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLabel" name="rippleRateLabel"/>
          </item>
          <item row="5" column="1" colspan="2">
            <widget class="QPushButton" name="updateParametersButton">
             <property name="text">
              <string>Update Parameters</string>
             </property>
            </widget>
          </item>
         </layout>
        </widget>
       </item>

       <item>
        <widget class="QGroupBox" name="rejectionParamsGroupBox">
         <property name="title">
          <string>False Positive Detection Parameters</string>
         </property>
         <layout class="QFormLayout">
          <item row="0" column="0" colspan="2">
           <widget class="QCheckBox" name="checkBox">
            <property name="text">
             <string>Enable Large Ripple Rejection</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Ripple Rejection Thresh. (Max)</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_2"/>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QCheckBox" name="checkBox_2">
            <property name="text">
             <string>Enable Noise Channels</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>[Noise] Ripple Band Threshold</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QDoubleSpinBox" name="noiseThreshold"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>[Noise] Num. Active Channels</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QSpinBox" name="noiseActiveChannels">
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="stimParamsGroupBox">
         <property name="title">
          <string>Special Stimulation Parameters</string>
         </property>
         <layout class="QFormLayout" name="formLayout_4">
          <item row="0" column="0" colspan="2">
           <widget class="QCheckBox" name="controlStimulationCheckBox">
            <property name="text">
             <string>Trigger ~200 ms Post Detection Delay Stimulation</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QCheckBox" name="checkBox_3">
            <property name="text">
             <string>Enable Extended Ripple Targeting</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Inter-Threshold Duration</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_5"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="enableStimulationButton">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>50</height>
          </size>
         </property>
         <property name="text">
          <string>Enable Stimulation</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>677</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
